### PR TITLE
Fixes #171: Added pool_utilization thresholds to nat source pools

### LINKED ISSUE #171: Feature request: Add pool-utilization-alarm to NAT source pool
This is a feature request to add pool-utilization-alarm to NAT source pool.

```
set security nat source pool poolName pool-utilization-alarm clear-threshold 60
set security nat source pool poolName pool-utilization-alarm raise-threshold 70
```
Junos reference:
[pool-utilization-alarm](https://www.juniper.net/documentation/us/en/software/junos/nat/topics/ref/statement/pool-utilization-alarm-edit-security-source-nat-pool.html)

I have the code ready. Please assign to me.
Thank you very much!

### PR DESCRIPTION
**Pull request by edpio19**
* add `pool-utilization-alarm` argument to `security_nat_source_pool` resource

### CODE REVIEW

**Review by jeremmfr — CHANGES_REQUESTED** [review]
A nested block `pool_utilization_alarm` just for 2 arguments is a bit too complex.
Can you rework the PR to propose only 2 new arguments ? : 
* `pool_utilization_alarm_raise_threshold`
* `pool_utilization_alarm_clear_threshold`

**Comment by edpio19** [comment]
@jeremmfr Changes have been made. Thank you for reviewing!

**Comment by jeremmfr** [comment]
> @jeremmfr Changes have been made. Thank you for reviewing!
The PR displays the commits already merged. Can you rebase correctly from main or recreate a new PR ?

**Comment by edpio19** [comment]
Thank you for pointing this out. It has been fixed now.

File: junos/resource_security_nat_source_pool.go [hunk 1]
```diff
@@ -282,6 +294,14 @@ func setSecurityNatSourcePool(d *schema.ResourceData, m interface{}, jnprSess *N
...
 		configSet = append(configSet, setPrefix+" routing-instance "+d.Get("routing_instance").(string))
 	}
+	if d.Get("pool_utilization_alarm_raise_threshold").(int) != 0 {
+		configSet = append(configSet, setPrefix+" pool-utilization-alarm raise-threshold "+
+			strconv.Itoa(d.Get("pool_utilization_alarm_raise_threshold").(int)))
+	}
+	if d.Get("pool_utilization_alarm_clear_threshold").(int) != 0 {
+		configSet = append(configSet, setPrefix+" pool-utilization-alarm clear-threshold "+
+			strconv.Itoa(d.Get("pool_utilization_alarm_clear_threshold").(int)))
+	}
```
Comment (jeremmfr) [inline junos/resource_security_nat_source_pool.go:304]: Can you insert the new options to respect resource (alphabetical) order ?

File: junos/resource_security_nat_source_pool.go [hunk 2]
```diff
@@ -324,6 +344,18 @@ func readSecurityNatSourcePool(natSourcePool string,
...
+					strings.TrimPrefix(itemTrim, "pool-utilization-alarm raise-threshold "))
+				if err != nil {
+					return confRead, fmt.Errorf("failed to convert value from '%s' to integer : %w", itemTrim, err)
+				}
+			case strings.HasPrefix(itemTrim, "pool-utilization-alarm clear-threshold "):
+				confRead.poolUtilizationAlarmClearThreshold, err = strconv.Atoi(
+					strings.TrimPrefix(itemTrim, "pool-utilization-alarm clear-threshold "))
+				if err != nil {
+					return confRead, fmt.Errorf("failed to convert value from '%s' to integer : %w", itemTrim, err)
+				}
```
Comment (jeremmfr) [inline junos/resource_security_nat_source_pool.go:358]: Can you insert the new options to respect resource (alphabetical) order ?

File: junos/resource_security_nat_source_pool.go [hunk 3]
```diff
@@ -359,6 +391,14 @@ func fillSecurityNatSourcePoolData(d *schema.ResourceData, natSourcePoolOptions
...
 		panic(tfErr)
 	}
+	if tfErr := d.Set("pool_utilization_alarm_raise_threshold",
+		natSourcePoolOptions.poolUtilizationAlarmRaiseThreshold); tfErr != nil {
+		panic(tfErr)
+	}
+	if tfErr := d.Set("pool_utilization_alarm_clear_threshold",
+		natSourcePoolOptions.poolUtilizationAlarmClearThreshold); tfErr != nil {
+		panic(tfErr)
+	}
```
Comment (jeremmfr) [inline junos/resource_security_nat_source_pool.go:401]: Can you insert the new options to respect resource (alphabetical) order ?

File: website/docs/r/security_nat_source_pool.html.markdown [hunk 4]
```diff
@@ -30,6 +30,8 @@ The following arguments are supported:
 * `port_overloading_factor` - (Optional)(`Int`) Port overloading factor for each IP.
 * `port_range` - (Optional)(`String`) Range of port for source nat.
 * `routing_instance` - (Optional)(`String`) Name of routing instance for switch with nat.
+* `pool_utilization_alarm_raise_threshold` - (Optional)(`Int`) Upper threshold at which an SNMP trap is triggered. Range 50 through 100.
+* `pool_utilization_alarm_clear_threshold` - (Optional)(`Int`) Lower threshold at which an SNMP trap is triggered. Range 40 through 100.
```
Comment (jeremmfr) [inline website/docs/r/security_nat_source_pool.html.markdown:34]: Can you insert the new options to respect resource (alphabetical) order ?

File: junos/resource_security_nat_source_pool.go [hunk 5]
```diff
@@ -65,6 +67,16 @@ func resourceSecurityNatSourcePool() *schema.Resource {
...
+			"pool_utilization_alarm_raise_threshold": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntBetween(50, 100),
+			},
+			"pool_utilization_alarm_clear_threshold": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntBetween(40, 100),
+			},
```
Comment (jeremmfr) [inline junos/resource_security_nat_source_pool.go:79]: Can you insert the new options to respect optional then alphabetical order ?
The `raise-threshold` is required with `clear-threshold` :
```suggestion
			"pool_utilization_alarm_raise_threshold": {
				Type:         schema.TypeInt,
				Optional:     true,
				ValidateFunc: validation.IntBetween(50, 100),
			},
			"pool_utilization_alarm_clear_threshold": {
				Type:         schema.TypeInt,
				Optional:     true,
				RequiredWith: []string{"pool_utilization_alarm_raise_threshold"},
				ValidateFunc: validation.IntBetween(40, 100),
			},
```

**Review by jeremmfr — CHANGES_REQUESTED** [review]
There are reordering to do and a `RequiredWith` option to add to clear-threshold argument

**Review by jeremmfr — APPROVED** [review]

**Comment by jeremmfr** [comment]
Looks good to me

**Pull request merged**